Showcase-iOSTests/About/View Model/AboutViewModelTests.swift

```diff
@@ -0,0 +1,116 @@
+        XCTAssertNotNil(viewModelUnderTest)
+    }
+
+    func testThatTwitterUrlHasTheExpectedValue() {
+        let viewModelUnderTest = AboutViewModel(socialMediaLinks: validSocialMediaResponse)
+        let actualResult = viewModelUnderTest.twitter
+        let expectedResult = URL(string: "https://twitter.com/dvt_corporate")
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testThatFacebookUrlHasTheExpectedValue() {
+        let viewModelUnderTest = AboutViewModel(socialMediaLinks: validSocialMediaResponse)
+        let actualResult = viewModelUnderTest.facebook
+        let expectedResult = URL(string: "https://www.facebook.com/DVTSoftware")
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testThatInstagramUrlHasTheExpectedValue() {
+        let viewModelUnderTest = AboutViewModel(socialMediaLinks: validSocialMediaResponse)
+        let actualResult = viewModelUnderTest.instagram
+        let expectedResult = URL(string: "https://www.instagram.com/dvtsoftware/")
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testThatWebsiteUrlHasTheExpectedValue() {
+        let viewModelUnderTest = AboutViewModel(socialMediaLinks: validSocialMediaResponse)
+        let actualResult = viewModelUnderTest.website
+        let expectedResult = URL(string: "https://www.dvt.co.za")
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+     // MARK: ViewModel Negative tests
+
+    func testThatTwitterUrlHasTheExpectedNilValue() {
+        let viewModelUnderTest = AboutViewModel(socialMediaLinks: invalidSocialMediaResponse)
+        let actualResult = viewModelUnderTest.twitter
+        let expectedResult = URL(string: "https://twitter.com/dvt_corporate")
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testThatFacebookUrlHasTheExpectedNilValue() {
+        let viewModelUnderTest = AboutViewModel(socialMediaLinks: invalidSocialMediaResponse)
+        let actualResult = viewModelUnderTest.facebook
+        let expectedResult = URL(string: "https://www.facebook.com/DVTSoftware")
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testThatInstagramUrlHasTheExpectedNilValue() {
+        let viewModelUnderTest = AboutViewModel(socialMediaLinks: invalidSocialMediaResponse)
+        let actualResult = viewModelUnderTest.instagram
+        let expectedResult = URL(string: "https://www.instagram.com/dvtsoftware/")
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testThatWebsiteUrlHasTheExpectedNilValue() {
+        let viewModelUnderTest = AboutViewModel(socialMediaLinks: invalidSocialMediaResponse)
+        let actualResult = viewModelUnderTest.website
+        let expectedResult = URL(string: "https://www.dvt.co.za")
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    // MARK: Mocking Firebase response dictionary
+
+    func mockValidSocialMediaResponse() -> [String: Any] {
```

Comment:
This seems to be duplicated in all your tests file, maybe we can put it in a class called GeneratedSocialMediaData and call these functions as static methods.

Reply:
done